### PR TITLE
ThreadPoolBudget.check() throws incorrect error messages #4070

### PR DESCRIPTION
**Pull request by cowwoc**
#4070

### CODE REVIEW

**Comment by sbordet** [comment]
@cowwoc branch `master` is unused. Use branch `jetty-9.4.x` instead.

**Comment by sbordet** [comment]
@cowwoc see also comment on the issue: we want at least `required+1` threads.

File: jetty-util/src/main/java/org/eclipse/jetty/util/thread/ThreadPoolBudget.java [hunk 1]
```diff
@@ -142,10 +142,10 @@ public void check() throws IllegalStateException
         int maximum = pool.getMaxThreads();
         int actual = maximum - required;
 
-        if (actual <= 0)
+        if (actual < 0)
         {
             infoOnLeases();
-            throw new IllegalStateException(String.format("Insufficient configured threads: required=%d < max=%d for %s", required, maximum, pool));
+            throw new IllegalStateException(String.format("Insufficient configured threads: max=%d < required=%d for %s", required, maximum, pool));
```
Comment (gregw) [inline jetty-util/src/main/java/org/eclipse/jetty/util/thread/ThreadPoolBudget.java]: You've swap the order of the format string to have max before required, but the arg order is still required then maximum

Comment (cowwoc) [reply jetty-util/src/main/java/org/eclipse/jetty/util/thread/ThreadPoolBudget.java]: Fixed. Thank you.

**Comment by gregw** [comment]
Not sure why CI is broken, but looks unrelated.
thanks

**Pull request merged**